### PR TITLE
Use hasContract when polling for moving to payments

### PR DESCRIPTION
**Pull request by vonElfvin**

### CODE REVIEW

File: src/pages/OfferNew/Checkout/index.tsx [hunk 1]
```diff
@@ -209,7 +206,7 @@ export const Checkout: React.FC<Props> = ({
       offerData.person.ssn,
   )
 
-  if (signStatus?.signState === SignState.Completed) {
+  if (hasContract) {
```
Comment (vonElfvin) [inline src/pages/OfferNew/Checkout/index.tsx:209]: We could also do this to make it super-safe
```suggestion
  if (signStatus?.signState === SignState.Completed && hasContract) {
```

Comment (vonElfvin) [reply src/pages/OfferNew/Checkout/index.tsx:209]: Where we check both the first and last step of the sign process. Right now creating the contract is last in the chain so to speak.

**Comment by vonElfvin** [comment]
> Cool! Just a general question, do we handle the state where sign is completed but you haven't received any contract yet? I haven't tried this but judging from the code it'd probably stop the loading state, but not redirect, right?

Very true, I now updated the loading state to also look at `hasContract`

File: src/pages/OfferNew/Checkout/index.tsx [hunk 2]
```diff
@@ -274,6 +275,7 @@ export const Checkout: React.FC<Props> = ({
             signQuotesMutation.loading ||
             signUiState === SignUiState.STARTED ||
             signUiState === SignUiState.STARTED_WITH_REDIRECT ||
+            !hasContract ||
```
Comment (palmenhq) [inline src/pages/OfferNew/Checkout/index.tsx:278]: won't this cause the loading state to always be true until you signed?

Comment (vonElfvin) [reply src/pages/OfferNew/Checkout/index.tsx]: Really good catch! Fixed now

**Comment by vonElfvin** [comment]
This will be solved by the backend instead. A similar outcome with a backend solution.

**Pull request closed without merging**